packages/amplify-graphql-api-construct/src/internal/construct-exports.ts

```diff
@@ -1,109 +1,163 @@
   }
 
-  const getL1Resource = <T extends CfnResource>({ stackName, id }: StackResourceMapping): T => {
-    const referencedStack = stackName ? transformerStack.getNestedStack(stackName).includedTemplate : transformerStack;
-    return referencedStack.getResource(id) as T;
-  };
+  if (!cfnGraphqlSchema) {
+    throw new Error('Expected to find AWS::AppSync::GraphQLSchema in the generated resource scope.');
+  }
 
-  const getL1Resources = <T extends CfnResource>(mappings: StackResourceMapping[]): Record<string, T> =>
-    Object.fromEntries(mappings.map((stackResourceMapping) => [stackResourceMapping.id, getL1Resource<T>(stackResourceMapping)]));
+  const nestedStacks: Record<string, NestedStack> = Object.fromEntries(
+    scope.node.children.filter(NestedStack.isNestedStack).map((nestedStack: NestedStack) => [nestedStack.node.id, nestedStack]),
+  );
 
   return {
-    cfnGraphqlApi: getL1Resource<CfnGraphQLApi>(apiResourceMapping),
-    cfnGraphqlSchema: getL1Resource<CfnGraphQLSchema>(schemaResourceMapping),
-    cfnApiKey: apiKeyResourceMapping && <CfnApiKey>getL1Resource(apiKeyResourceMapping),
-    cfnResolvers: getL1Resources<CfnResolver>(cfnResolverResourceMapping),
-    cfnFunctionConfigurations: getL1Resources<CfnFunctionConfiguration>(cfnAppSyncFunctionResourceMapping),
-    cfnDataSources: getL1Resources<CfnDataSource>(cfnDataSourceResourceMapping),
-    cfnTables: getL1Resources<CfnTable>(cfnTableResourceMapping),
-    cfnRoles: getL1Resources<CfnRole>(cfnRoleResourceMapping),
-    cfnPolicies: getL1Resources<CfnPolicy>(cfnPolicyResourceMapping),
-    additionalCfnResources: getL1Resources(additionalResourceMapping),
+    graphqlApi: GraphqlApi.fromGraphqlApiAttributes(scope, 'L2GraphqlApi', { graphqlApiId: cfnGraphqlApi.attrApiId }),
+    tables,
+    roles,
+    functions,
+    nestedStacks,
+    cfnResources: {
+      cfnGraphqlApi,
+      cfnGraphqlSchema,
+      cfnApiKey,
+      cfnResolvers,
+      cfnFunctionConfigurations,
+      cfnDataSources,
+      cfnTables,
+      cfnRoles,
+      cfnFunctions,
+      additionalCfnResources,
+    },
   };
 };
+
+/**
+ * Get the function slots generated by the Graphql transform operation, adhering to the FunctionSlot interface.
+ * @param generatedResolvers the resolvers generated by the transformer to spit back out.
+ * @returns the list of generated function slots in the transformer, in order to facilitate overrides.
+ */
+export const getGeneratedFunctionSlots = (generatedResolvers: Record<string, string>): FunctionSlot[] =>
+  Object.entries(generatedResolvers)
+    .filter(([name]) => name.split('.').length === 6)
+    .map(([name, resolverCode]) => {
+      const [typeName, fieldName, slotName, slotIndex, templateType] = name.split('.');
+      return {
+        typeName,
+        fieldName,
+        slotName,
+        slotIndex: Number.parseInt(slotIndex, 10),
+        function: {
+          // TODO: this should consolidate req/req values back together
```

Comment:
is this any concerning TODO that we left unintentionally?

Reply:
Oh no, I meant to strip the todo, but must have reverted. It's a DX improvement for the CDK construct that's covered by an existing piece of QA.